src/svlog/typeck.rs

```diff
@@ -670,6 +670,13 @@ pub(crate) fn packed_type_from_ast<'a>(
             packed_type_from_def(cx, def, name.span, env)
         }
 
+        // Forward declarations
+        ast::ForwardType { kind: _ } => {
+            // TODO This should likely be a nicer error if we get here because a forward
+            // declaration is not later defined.
+            bug_span!(ast.span(), cx, "type {:#1?} not implemented", ast.kind)
+        }
+
```

Comment:
I think it's okay to just trigger a `bug_span!` here: Technically the `ast::ForwardType` marker should never make its way into `packed_type_from_ast` via any other means than type checking of a `typedef`. I'd rather suggest we add some code to the type computation of a `typedef` (one of the callers of this `packed_type_from_ast`) and -- if we detect a forward declaration -- issue a nice error about no definition having been provided. Let's do something like that in a future PR.